step.sh

```diff
@@ -54,14 +54,21 @@ fi
 TAG_NAME=""
 
 if [ -z "$bitrise_tag_format" ]; then
-	printf -v TAG_NAME "v%s(%s)" "$CFBundleShortVersionString" "$CFBundleVersion"
+    printf -v TAG_NAME "v%s(%s)" "$CFBundleShortVersionString" "$CFBundleVersion"
 else
-	printf -v TAG_NAME "$bitrise_tag_format" "$CFBundleShortVersionString" "$CFBundleVersion"
+    name="${bitrise_tag_format//_VERSION_/$CFBundleShortVersionString}"
+    name="${name//_BUILD_/$CFBundleVersion}"
+    printf -v TAG_NAME "$name"
 fi
 echo $TAG_NAME
 git checkout "$BITRISE_GIT_BRANCH"
-git tag "$TAG_NAME"
-git push --tags
+git tag "$TAG_NAME" "$GIT_CLONE_COMMIT_HASH"
+
+if [[ $use_lightweight_tag = "yes" ]]; then
+    git push origin "$TAG_NAME"
```

Comment:
Wouldn't [this scenario](https://discuss.bitrise.io/t/intermittent-build-failures-to-execute-push-to-git-remote-how-to-manage-build-number/382/2) happen in this case?

Reply:
I don't think it would, as we do a checkout of the branch, not of a commit hash: 
 `git checkout "$BITRISE_GIT_BRANCH"`

Reply:
Got it👍